Description: Fix concurrency count determination on Debian GNU/kFreeBSD
Author: Jelmer Vernooij <jelmer@debian.org>
Status: Submitted for inclusion upstream, https://code.launchpad.net/~jelmer/bzr/cpu-count-bsd/+merge/81458

=== modified file 'bzrlib/osutils.py'
--- a/bzrlib/osutils.py	2011-10-12 16:00:13 +0000
+++ b/bzrlib/osutils.py	2011-11-07 15:08:19 +0000
@@ -2323,7 +2323,14 @@
             except (OSError, IOError):
                 pass
         else:
-            concurrency = multiprocessing.cpu_count()
+            # multiprocessing.cpu_count() isn't implemented on all platforms
+            try:
+                concurrency = multiprocessing.cpu_count()
+            except NotImplementedError:
+                try:
+                    concurrency = _local_concurrency()
+                except (OSError, IOError):
+                    pass
     try:
         concurrency = int(concurrency)
     except (TypeError, ValueError):

